core/objects.go

```diff
@@ -254,6 +274,10 @@ type Challenge struct {
 
 	// Used by dns and dvsni challenges
 	Validation *jose.JsonWebSignature `json:"validation,omitempty"`
+
+	// Contains information about URLs used or redirected to and IPs resolved and
+	// used
+	ValidationRecord *ValidationRecord `json:"validationRecord,omitempty"`
```

Comment:
This is at the request of @jsha ([here](https://github.com/letsencrypt/boulder/pull/505#discussion_r36156776)), my original implementation used a single shared object.